src/code.c

```diff
@@ -201,6 +180,24 @@ SEXP method_call_(SEXP call, SEXP generic, SEXP envir) {
     R7_method_lookup_error(generic, signature_classes, envir);
   }
 
+  // Now add the remaining, non-dispatched, args
+  SEXP gen_formals = FORMALS(generic);
+  R_xlen_t gen_formals_len = Rf_xlength(gen_formals);
+  if (gen_formals_len > gen_signature_len) {
+    for (R_xlen_t i = gen_signature_len; i < gen_formals_len; ++i) {
+      // Find value of that argument
+      SEXP name = Rf_install(CHAR(STRING_ELT(gen_signature_args, i)));
+      SEXP arg = Rf_findVar(name, envir);
+
+      // And add to args
+      SETCDR(tail, Rf_cons(arg, R_NilValue));
+      tail = CDR(tail);
```

Comment:
Do we need to set the `TAG()` here too? I don't think the argument names matter for the dispatch argument (since they're always in the correct order), but I think it will matter here.